src/acquisition/flusurv/flusurv_update.py

```diff
@@ -113,38 +112,185 @@ def update(issue, location_name, test_mode=False):
+        #     {'Variable': 'Age', 'valueid': 98, 'parentid': 0, 'Label': '>= 18', 'Color_HexValue': '#000000', 'Enabled': True},
+        #
+        #     {'Variable': 'Race', 'valueid': 1, 'parentid': None, 'Label': 'White', 'Color_HexValue': '#516889', 'Enabled': True},
+        #     {'Variable': 'Race', 'valueid': 2, 'parentid': None, 'Label': 'Black', 'Color_HexValue': '#44b3c6', 'Enabled': True},
+        #     {'Variable': 'Race', 'valueid': 3, 'parentid': None, 'Label': 'Hispanic/Latino', 'Color_HexValue': '#d19833', 'Enabled': True},
+        #     {'Variable': 'Race', 'valueid': 4, 'parentid': None, 'Label': 'Asian/Pacific Islander', 'Color_HexValue': '#cc5e56', 'Enabled': True},
+        #     {'Variable': 'Race', 'valueid': 5, 'parentid': None, 'Label': 'American Indian/Alaska Native', 'Color_HexValue': '#007d8e', 'Enabled': True},
+        #
+        #     {'Variable': 'Sex', 'valueid': 1, 'parentid': None, 'Label': 'Male', 'Color_HexValue': '#44b3c6', 'Enabled': True},
+        #     {'Variable': 'Sex', 'valueid': 2, 'parentid': None, 'Label': 'Female', 'Color_HexValue': '#F2775F', 'Enabled': True},
+        #
+        #     {'Variable': None, 'valueid': 0, 'parentid': 0, 'Label': 'Overall', 'Color_HexValue': '#000000', 'Enabled': True},
+        # ]
+        #
+        # All 23 strata are available starting with epiweek 200935.
+        #
+        # The previous version of the GRASP API
+        #  (https://gis.cdc.gov/GRASP/Flu3/GetPhase03InitApp)
+        #  used the following age groupid mapping, as described in the
+        #  top-level "ages" element:
+        #    'ages' = [
+        #        {'label': '0-4 yr', 'ageid': 1, 'color_hexvalue': '#1B9E77'},
+        #        {'label': '5-17 yr', 'ageid': 2, 'color_hexvalue': '#D95F02'},
+        #        {'label': '18-49 yr', 'ageid': 3, 'color_hexvalue': '#4A298B'},
+        #        {'label': '50-64 yr', 'ageid': 4, 'color_hexvalue': '#E7298A'},
+        #        {'label': '65+ yr', 'ageid': 5, 'color_hexvalue': '#6AA61E'},
+        #        {'label': 'Overall', 'ageid': 6, 'color_hexvalue': '#000000'},
+        #        {'label': '65-74 yr', 'ageid': 7, 'color_hexvalue': '#A6CEE3'},
+        #        {'label': '75-84 yr', 'ageid': 8, 'color_hexvalue': '#CAB2D6'},
+        #        {'label': '85+', 'ageid': 9, 'color_hexvalue': '#1f78b4'}
+        #    ]
+        #
+        #  In addition to the new age, race, and sex breakdowns, the group
+        #  id for overall reporting has changed from 6 to 0. Ageids 1-5
+        #  and 7-9 retain the same the same meanings.
+        n_expected_groups = 23
+        # Subtract one since we also store the season description in each epiweek value
+        if len(data[epiweek].keys() - 1) != n_expected_groups:
+            warnings.warn(
+                f"{location} {epiweek} data does not contain the expected {n_expected_groups} groups"
```

Comment:
for ease of debugging problems we might see in production, print the number of groups seen, and possibly even the group names themselves.

you may want to differentiate between <23 groups (implying some are missing (would that be a fatal error in the `cur.execute()` call below, if a named value isnt in the params list?)) and >23 groups (implying new groups have been added, which we might want to investigate and eventually add to the db).

if you want to ***really*** over engineer this and super future-proof it, you could save the set of 23 expected keys instead of just the count, and print seen keys minus expected as well as expected keys minus seen.  that might be helpful in the possible event that groups are renamed (but the count will still be 23).